Firestore/core/test/firebase/firestore/remote/serializer_test.cc

```diff
@@ -454,19 +454,45 @@ TEST_F(SerializerTest, EncodesTimestamps) {
 }
 
 TEST_F(SerializerTest, EncodesBlobs) {
-  std::vector<std::vector<uint8_t>> cases{
+  std::vector<ByteString> cases{
       {},
       {0, 1, 2, 3},
       {0xff, 0x00, 0xff, 0x00},
   };
 
-  for (const std::vector<uint8_t>& blob_value : cases) {
-    FieldValue model =
-        FieldValue::FromBlob(blob_value.data(), blob_value.size());
+  for (const ByteString& blob_value : cases) {
+    FieldValue model = FieldValue::FromBlob(blob_value);
     ExpectRoundTrip(model, ValueProto(blob_value), FieldValue::Type::Blob);
   }
 }
 
+TEST_F(SerializerTest, EncodesNullBlobs) {
+  ByteString blob;
+  FieldValue model = FieldValue::FromBlob(blob);
+  ASSERT_EQ(blob.get(), nullptr);
```

Comment:
Optional: perhaps move this assertion one line up? As it stands, it seems to check that `blob` _became_ null after the creation of a field value.